lib/ansible/modules/monitoring/zabbix_map.py

```diff
@@ -0,0 +1,800 @@
+# -*- coding: utf-8 -*-
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: zabbix_map
+author:
+    - "Anton Alekseyev (@Akint)"
+short_description: Zabbix map creates/updates/deletes
+description:
+    - "This module allows you to create, modify and delete Zabbix map entries, using DOT language.
+      Nodes of the graph will become map elements and edges will become links between map elements.
+      See U(https://en.wikipedia.org/wiki/DOT_(graph_description_language)) for details.
+      Inspired by U(http://blog.zabbix.com/maps-for-the-lazy/)."
+    - "The following extra node attributes are supported:
+        C(zbx_host) contains name of the host in Zabbix. Use this if desired type of map element is C(host).
+        C(zbx_group) contains name of the host group in Zabbix. Use this if desired type of map element is C(host group).
+        C(zbx_map) contains name of the map in Zabbix. Use this if desired type of map element is C(map).
+        C(zbx_label) contains label of map element.
+        C(zbx_image) contains name of the image used to display the element in default state.
+        C(zbx_image_disabled) contains name of the image used to display disabled map element.
+        C(zbx_image_maintenance) contains name of the image used to display map element in maintenance.
+        C(zbx_image_problem) contains name of the image used to display map element with problems.
+        C(zbx_url) contains map element URL in C(name:url) format.
+            More than one url could be specified by adding some prefix (e.g., C(zbx_url1), C(zbx_url2))."
+    - "The following extra link attributes are supported:
+        C(zbx_draw_style) contains link line draw style. Possible values: C(line), C(bold), C(dotted), C(dashed).
+        C(zbx_trigger) contains name of the trigger used as a link indicator in C(host_name:trigger_name) format.
+            More than one trigger could be specified by adding some prefix (e.g., C(zbx_trigger1), C(zbx_trigger2)).
+        C(zbx_trigger_color) contains indicator color specified either as CSS3 name or as a hexadecimal code starting with C(#).
+        C(zbx_trigger_draw_style) contains indicator draw style. Possible values are the same as for C(zbx_draw_style)."
+requirements:
+    - "python >= 2.6"
+    - zabbix-api
+    - pydotplus
+    - webcolors
+    - PIL
```

Comment:
PIL is not available in Python 3, and ansible is compatible with Python 3.5+.

Pillow should be a drop-in replacement.